packages/react/src/transactions/adapters/useUnfollowController.ts

```diff
@@ -0,0 +1,38 @@
+import {
+  PendingSigningRequestError,
+  WalletConnectionError,
+  UserRejectedError,
+} from '@lens-protocol/domain/entities';
+import { UnfollowProfile, UnfollowRequest } from '@lens-protocol/domain/use-cases/profile';
+import { IEquatableError } from '@lens-protocol/shared-kernel';
+
+import { useSharedDependencies } from '../../shared';
+import { PromiseResultPresenter } from './PromiseResultPresenter';
+import { UnfollowProfileCallGateway } from './UnfollowProfileCallGateway';
+
+export function useUnfollowController() {
+  const { activeWallet, apolloClient, transactionGateway, protocolCallRelayer, transactionQueue } =
+    useSharedDependencies();
+
+  return async (request: UnfollowRequest) => {
+    const presenter = new PromiseResultPresenter<
+      void,
+      PendingSigningRequestError | WalletConnectionError | UserRejectedError | IEquatableError
+    >();
+
+    const unfollowProfileCallGateway = new UnfollowProfileCallGateway(apolloClient);
+
+    const unfollowProfiles = new UnfollowProfile(
+      activeWallet,
+      transactionGateway,
+      unfollowProfileCallGateway,
+      protocolCallRelayer,
+      transactionQueue,
+      presenter,
+    );
+
+    void unfollowProfiles.execute(request);
```

Comment:
seems we can just `await` here

Reply:
I think this is true for most operations... the release of the current call stack for any operation that ends up involving the `TransactionQueue` is done within the `TransactionQueue` itself. Awaiting this promise is a great opportunity to let invariant errors bubble up to the call site, so the developer can act on coding mistakes or bugs involving their setup.

Reply:
thanks yeah, this is good find. added in the `await` 👍

Reply:
did you? I still can see `void` there 🤔